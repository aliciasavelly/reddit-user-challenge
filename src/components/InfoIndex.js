import React from 'react';

const sortByScore = (a, b) => {
	if (a.score > b.score) {
		return -1;
	} else if (a.score < b.score) {
		return 1;
	} else {
		return 0;
	}
};

const InfoIndex = ({ index, shownInfo, username }) => {

	if (index.length === 0) {
		return <div className="no-info">No data to show!</div>;
	} else {
		return(
			<div>
				<div className="title">
					{`${username}'s ${shownInfo}`}
				</div>
				{
					index.sort(sortByScore).map((info, i) => {
						const displayedInfo = info.body || info.title;
						return(
							<div className="info-wrapper" key={info + i}>
								<div className="score">{info.score}</div>
								<div className="info">
									<div className="displayed-info">
										{ displayedInfo }
									</div>
									<a 
										href={info.url} 
										target="_blank" 
										className="url"
									>
										{info.url}
									</a>
								</div>
							</div>
						);
					})
				}
			</div>
		);
	}
};

export default InfoIndex;