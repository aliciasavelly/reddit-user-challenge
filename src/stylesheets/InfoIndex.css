.title {
	text-align: center;
	padding: 20px;
	font-weight: bold;
}

.info-wrapper {
	border-top: 1px solid gray;
	padding: 40px;
	display: flex;
}

.displayed-info {
	font-weight: bold;
	margin-bottom: 10px;
}

.no-info {
	padding: 40px;
	text-align: center;
}

.score {
	margin-right: 15px;
	font-weight: bold;
	color: gray;
}
