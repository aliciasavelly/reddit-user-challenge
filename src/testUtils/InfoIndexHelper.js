export const postIndex = [
	{
		title: "Post 1",
		score: 5,
		url: "www.reddit.com/1",
	},
	{ title: "Post 2", 
		score: 7,
		url: "www.reddit.com/2",
	}
];

export const commentIndex = [
	{
		body: "Comment 1",
		score: -2,
		url: "www.reddit.com/comment/1",
	},
	{
		body: "Comment 2",
		score: -4,
		url: "www.reddit.com/comment/2",
	}
]