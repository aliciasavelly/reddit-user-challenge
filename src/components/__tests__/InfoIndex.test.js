import React from 'react';
import ReactDOM from 'react-dom';
import InfoIndex from '../InfoIndex';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { postIndex, commentIndex } from '../../testUtils/InfoIndexHelper.js';

describe('InfoIndex', () => {
	it('it renders the post index', () => {
		const renderedIndex = shallow(<InfoIndex index={postIndex} />);
		expect(shallowToJson(renderedIndex)).toMatchSnapshot();
	});

	it ('it renders the comment index', () => {
		const renderedIndex = shallow(<InfoIndex index={commentIndex} />);
		expect(shallowToJson(renderedIndex)).toMatchSnapshot();
	});

	it("it renders 'no data' if index is empty", () => {
		const index = [];
		const renderedIndex = shallow(<InfoIndex index={index} />);
		expect(shallowToJson(renderedIndex)).toMatchSnapshot();
	});
});



