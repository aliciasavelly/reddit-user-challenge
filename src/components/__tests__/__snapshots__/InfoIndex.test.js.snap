// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoIndex it renders 'no data' if index is empty 1`] = `
<div
  className="no-info"
>
  No data to show!
</div>
`;

exports[`InfoIndex it renders the comment index 1`] = `
<div>
  <div
    className="title"
  >
    undefined's undefined
  </div>
  <div
    className="info-wrapper"
  >
    <div
      className="score"
    >
      -2
    </div>
    <div
      className="info"
    >
      <div
        className="displayed-info"
      >
        Comment 1
      </div>
      <a
        className="url"
        href="www.reddit.com/comment/1"
        target="_blank"
      >
        www.reddit.com/comment/1
      </a>
    </div>
  </div>
  <div
    className="info-wrapper"
  >
    <div
      className="score"
    >
      -4
    </div>
    <div
      className="info"
    >
      <div
        className="displayed-info"
      >
        Comment 2
      </div>
      <a
        className="url"
        href="www.reddit.com/comment/2"
        target="_blank"
      >
        www.reddit.com/comment/2
      </a>
    </div>
  </div>
</div>
`;

exports[`InfoIndex it renders the post index 1`] = `
<div>
  <div
    className="title"
  >
    undefined's undefined
  </div>
  <div
    className="info-wrapper"
  >
    <div
      className="score"
    >
      7
    </div>
    <div
      className="info"
    >
      <div
        className="displayed-info"
      >
        Post 2
      </div>
      <a
        className="url"
        href="www.reddit.com/2"
        target="_blank"
      >
        www.reddit.com/2
      </a>
    </div>
  </div>
  <div
    className="info-wrapper"
  >
    <div
      className="score"
    >
      5
    </div>
    <div
      className="info"
    >
      <div
        className="displayed-info"
      >
        Post 1
      </div>
      <a
        className="url"
        href="www.reddit.com/1"
        target="_blank"
      >
        www.reddit.com/1
      </a>
    </div>
  </div>
</div>
`;
